Fix StrictlyBlittable check to work for "source Compilation" references.

jkoritzinsky opened the pull request:
Fixes #84739.

jkoritzinsky committed:
Fix StrictlyBlittable check to work for "source Compilation" references.

Fixes #84739.

AaronRobinsonMSFT commented on src/libraries/System.Runtime.InteropServices/gen/Microsoft.Interop.SourceGeneration/TypeSymbolExtensions.cs:
```diff
@@ -30,10 +30,10 @@ public static bool IsConsideredBlittable(this ITypeSymbol type)
             unsafe
             {
-                return IsBlittableWorker(type, ImmutableHashSet.Create<ITypeSymbol>(SymbolEqualityComparer.Default), &IsConsideredBlittableWorker);
+                return IsBlittableWorker(type, ImmutableHashSet.Create<ITypeSymbol>(SymbolEqualityComparer.Default), compilation: null, &IsConsideredBlittableWorker);
```
I thought we had nullability checks enabled. Did we turn them off?

jkoritzinsky replied:
They've been acting really weird ever since we moved into dotnet/runtime. I'll see if I can get them re-enabled.

jkoritzinsky replied:
dotnet/runtime libraries adds `<NoWarn>$(NoWarn);nullable</NoWarn>` to any project that doesn't target .NETCoreApp.

Looks like all other source generators in the repo multi-target to get nullable warnings and ensure the warnings from the BCL types are accurate (and then only ship the .NET Standard 2.0 assembly)

jkoritzinsky committed:
Be a better citizen around nullability

jkoritzinsky committed:
Explicitly treat pointers as blittable in our strict-blittability check. This is required now that we're being more exact on our check.

Beforehand, we were allowing through types that don't have a containing assembly (constructed types that aren't "named types"), which is more general than we want.
The main change in this PR forces us to be better here on our check.
